src/saga/posts.js

```diff
@@ -1,25 +1,43 @@
 import { put, takeEvery, call } from "redux-saga/effects";
-import { FETCH_POSTS, setPosts } from "../store/postsReducer";
+import {
+  FETCH_POSTS,
+  setPosts,
+  setPostsCount,
+  FETCH_POSTS_BY_PAGE,
+} from "../store/postsReducer";
+import { LOADED_POSTS, LOADING_POSTS } from "../store/loadingReducer";
+import { getAllPosts, getPostByPage } from "../apiServices";
 
-const fetchPostsFromApi = () =>
-  fetch("https://jsonplaceholder.typicode.com/posts");
+const requestPostsError = () => {
+  return { type: "REQUESTED_POSTS_FAILED" };
+};
 
-  const requestPostsError = () => {
-    return { type: 'REQUESTED_POSTS_FAILED' }
-  };
-  
+const loading = () =>
+  new Promise((resolve, reject) => {
+    setTimeout(resolve, 5000);
+  });
 
 function* fetchPostsWorker() {
-    try{
-        const data = yield call(fetchPostsFromApi);
-        const json = yield call(() => new Promise((res) => res(data.json())));
-        yield put(setPosts(json));
+  try {
+    yield put({ type: LOADING_POSTS, payload: "true" });
+    const postsCount = yield call(getAllPosts);
+    const post = yield call(getPostByPage, 1, 10);
+    yield loading();
+    yield put({ type: LOADED_POSTS, payload: "false" });
```

Comment:
используй boolean тип:
```suggestion
    yield put({ type: LOADED_POSTS, payload: false });
```